src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3402,9 +3402,9 @@ private Message GetSingleStreamReadMessage(RedisKey key, RedisValue afterId, int
 
         private Message GetStreamTrimMessage(RedisKey key, int maxLength, bool useApproximateMaxLength, CommandFlags flags)
         {
-            if (maxLength <= 0)
+            if (maxLength < 0)
             {
-                throw new ArgumentOutOfRangeException(nameof(maxLength), "maxLength must be greater than 0.");
+                throw new ArgumentOutOfRangeException(nameof(maxLength), "maxLength must be equal or greater than 0.");
```

Comment:
One tweak then good to go!
```suggestion
                throw new ArgumentOutOfRangeException(nameof(maxLength), "maxLength must be equal to or greater than 0.");
```